### PR TITLE
MINOR: Fix broken standalone ReplicationQuotasTestRig test

### PR DESCRIPTION
**Pull request by omkreddy**
-  Fix ZkUtils. getReassignmentJson to return valid  json string
-  Allow new file creation in ReplicationQuotasTestRig test

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by omkreddy** [comment]
@ijuma  Please take a look when you get a chance

File: core/src/main/scala/kafka/utils/ZkUtils.scala [hunk 1]
```diff
@@ -169,7 +169,7 @@ object ZkUtils {
   }
 
   def getReassignmentJson(partitionsToBeReassigned: Map[TopicPartition, Seq[Int]]): String = {
-    Json.encodeAsString(Map(
+    Json.legacyEncodeAsString(Map(
```
Comment (ijuma) [inline core/src/main/scala/kafka/utils/ZkUtils.scala]: Why is this needed?

Comment (omkreddy) [reply core/src/main/scala/kafka/utils/ZkUtils.scala]: when we use Json.encodeAsString,  Scala Map is getting encoded as ```{"empty":false,"traversableAgain":true}```. 

Related link: https://stackoverflow.com/questions/27948128/how-to-convert-scala-map-into-json-string

Otherwise, we need to pass Scala Map as Java Map.

Comment (ijuma) [reply core/src/main/scala/kafka/utils/ZkUtils.scala]: We should pass a Java map.

**Review by ijuma — APPROVED** [review]
LGTM

**Comment by ijuma** [comment]
Merged to trunk and 2.1.
